pavelib/paver_tests/test_database.py

```diff
@@ -59,3 +64,141 @@ def test_extract_files_from_zip(self, _mock_verify):
         with open(extracted_file, 'r') as test_file:
             data = test_file.read()
         assert data == 'Test file content'
+
+
+def _write_temporary_db_cache_files(path, files):
+    """
+    create some temporary files to act as the local db cache files so that
+    we can compute a fingerprint
+    """
+    for index, filename in enumerate(files):
+        filepath = os.path.join(path, filename)
+        with open(filepath, 'w') as cache_file:
+            cache_file.write(str(index))
+
+
+class TestPaverDatabaseTasks(MockS3Mixin, TestCase):
+    """
+    Tests for the high level database tasks
+    """
+
+    def setUp(self):
+        super(TestPaverDatabaseTasks, self).setUp()
+        conn = boto.connect_s3()
+        conn.create_bucket('moto_test_bucket')
+        self.bucket = conn.get_bucket('moto_test_bucket')
+
+    @patch.object(db_utils, 'CACHE_FOLDER', mkdtemp())
+    @patch.object(db_utils, 'FINGERPRINT_FILEPATH', os.path.join(mkdtemp(), 'fingerprint'))
+    @patch.object(db_utils, 'sh')
+    def test_load_data_from_local_cache(self, _mock_sh):
+        """
+        Assuming that the computed db cache file fingerprint is the same as
+        the stored fingerprint, verify that we make a call to load data into
+        the database without running migrations
+        """
+        self.addCleanup(shutil.rmtree, db_utils.CACHE_FOLDER)
```

Comment:
nit: seems like a bunch of this setup code is repeated and could be moved to the setUp method. We might want to refactor in a future PR.